mp4/sgpd.go

```diff
@@ -0,0 +1,134 @@
+package mp4
+
+import (
+	"io"
+	"io/ioutil"
+)
+
+// SgpdBox - Sample Group Description Box, ISO/IEC 14496-12 (2020) 8.9.3
+// Version 0 is deprecated
+type SgpdBox struct {
+	Version                      byte
+	Flags                        uint32
+	GroupingType                 string // uint32, but takes values such as seig
+	DefaultLength                uint32
+	DefaultGroupDescriptionIndex uint32
+	DescriptionLengths           []uint32
+	SampleGroupEntries           []SampleGroupEntry
+}
+
+// DecodeSgpd - box-specific decode
+func DecodeSgpd(hdr *boxHeader, startPos uint64, r io.Reader) (Box, error) {
+	data, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	s := NewSliceReader(data)
+	versionAndFlags := s.ReadUint32()
+	version := byte(versionAndFlags >> 24)
+
+	b := &SgpdBox{
+		Version: version,
+		Flags:   versionAndFlags & flagsMask,
+	}
+	b.GroupingType = s.ReadFixedLengthString(4)
+
+	if b.Version >= 1 {
+		b.DefaultLength = s.ReadUint32()
+	}
+	if b.Version >= 2 {
+		b.DefaultGroupDescriptionIndex = s.ReadUint32()
+	}
+	entryCount := int(s.ReadUint32())
+	for i := 0; i < entryCount; i++ {
+		var descriptionLength uint32 = b.DefaultLength
+		if b.Version >= 1 && b.DefaultLength == 0 {
```

Comment:
Same here:
```
 if (version==1) {
    if (default_length==0) {
       unsigned int(32) description_length;
    }
 } 
```

Reply:
This is again according to the 2020 spec.